signer/src/storage/mod.rs

```diff
@@ -226,12 +226,26 @@ pub trait DbRead {
     where
         X: Into<PublicKeyXOnly> + Send;
 
+    /// Return the applicable DKG shares status for the given aggregate key
+    fn get_dkg_shares_status<X>(
+        &self,
+        aggregate_key: X,
+    ) -> impl Future<Output = Result<Option<model::DkgSharesStatus>, Error>> + Send
+    where
+        X: Into<PublicKeyXOnly> + Send;
```

Comment:
We could define a default impl using `get_encrypted_dkg_shares`: it would not be as optimized query-wise, but it would be one less query (and reusing one that it's already used/tested around) and it's not like fetching a full row it's our db issue at the moment 😆

Reply:
Yeah that works, I'll do that and remove the new query.

Reply:
Cleaned up in https://github.com/stacks-network/sbtc/pull/1318/commits/a2f5b6b7816284955840efff7c90b69ca402e022